core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -213,6 +213,16 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
  private:
   /** \brief finalize chunk_size if it was set to AUTO*/
   inline void set_auto_chunk_size() {
+#ifdef KOKKOS_ENABLE_SYCL
+    if (std::is_same_v<typename traits::execution_space,
```

Comment:
Possibly:
```suggestion
    if constexpr (std::is_same_v<typename traits::execution_space,
```
(makes it explicit that this is a static check)

Reply:
Hmm... I don't see much of an improvement. I'll change it if someone requests it, otherwise, I would leave it as is.